File: ga/glm4_result_2024_6/result/1968B_glm/24_1968B.c
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: i32 = input.trim().parse().expect("Please enter a number");

    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let mut parts = input.trim().split_whitespace();
        let n: usize = parts.next().expect("Please enter a number").parse().expect("Please enter a valid number");
        let m: usize = parts.next().expect("Please enter a number").parse().expect("Please enter a valid number");
        let sn = parts.next().expect("Please enter a string").chars().take(n).collect::<String>();
        let sm = parts.next().expect("Please enter a string").chars().take(m).collect::<String>();

        let mut i = 0;
        let mut j = 0;
        while i < sn.len() && j < sm.len() {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}

File: ga/4omini_result_2024_7/result/1968B_cpp_mini/24_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let mut parts = input.split_whitespace();
        let x: i32 = parts.next().unwrap().parse().unwrap();
        let k: i32 = parts.next().unwrap().parse().unwrap();

        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let a: String = input.trim().to_string();
        
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let b: String = input.trim().to_string();

        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.as_bytes()[ans] == i as u8 {
                ans += 1;
            }
        }
        println!("{}", ans);
    }
}